pybop/optimisers/scipy_optimisers.py

```diff
@@ -111,79 +208,80 @@ def name(self):
     """
 
-    def __init__(
-        self, bounds=None, strategy="best1bin", maxiter=1000, popsize=15, tol=1e-5
-    ):
-        super().__init__()
-        self.tol = tol
-        self.strategy = strategy
-        self._max_iterations = maxiter
-        self._population_size = popsize
+    def __init__(self, cost, **optimiser_kwargs):
+        self.__dict__.update(DEFAULT_SCIPY_DIFFERENTIAL_EVOLUTION_OPTIONS)
+        super().__init__(cost, **optimiser_kwargs)
 
-        if bounds is None:
-            raise ValueError("Bounds must be specified for differential_evolution.")
-        elif not all(
-            np.isfinite(value) for sublist in bounds.values() for value in sublist
-        ):
-            raise ValueError("Bounds must be specified for differential_evolution.")
-        elif isinstance(bounds, dict):
-            bounds = [
-                (lower, upper) for lower, upper in zip(bounds["lower"], bounds["upper"])
-            ]
-        self.bounds = bounds
-
-    def _runoptimise(self, cost_function, x0=None):
+    def _run_optimiser(self):
         """
         Executes the optimization process using SciPy's differential_evolution function.
 
-        Parameters
-        ----------
-        cost_function : callable
-            The objective function to minimize.
-        x0 : array_like, optional
-            Ignored parameter, provided for API consistency.
-
         Returns
         -------
         tuple
-            A tuple (x, final_cost) containing the optimized parameters and the value of ``cost_function`` at the optimum.
+            A tuple (x, final_cost) containing the optimized parameters and the value of
+            the cost function at the optimum.
         """
-
-        self.log = []
-
-        if x0 is not None:
+        if self.x0 is not None:
             print(
                 "Ignoring x0. Initial conditions are not used for differential_evolution."
             )
+        self.x0 = None
 
         # Add callback storing history of parameter values
         def callback(x, convergence):
             self.log.append([x])
 
+        # Reformat bounds
+        if self.bounds is None:
+            raise ValueError("Bounds must be specified for differential_evolution.")
+        elif isinstance(self.bounds, dict):
+            if not all(
+                np.isfinite(value)
+                for sublist in self.bounds.values()
+                for value in sublist
+            ):
+                raise ValueError("Bounds must be specified for differential_evolution.")
+            bounds = [
+                (lower, upper)
+                for lower, upper in zip(self.bounds["lower"], self.bounds["upper"])
+            ]
+        else:
+            if not all(np.isfinite(value) for pair in self.bounds for value in pair):
+                raise ValueError("Bounds must be specified for differential_evolution.")
+            bounds = self.bounds
```

Comment:
I think this can be completed so that bounds is retained as a class attribute. I would improve readability if we retain `self.bounds` by either modifying inplace or creating a copy. We can do this in a separate PR if however.